src/shared/dissect-image.h

```diff
@@ -110,6 +110,7 @@ typedef enum DissectImageFlags {
         DISSECT_IMAGE_MOUNT_READ_ONLY     = 1 << 17, /* Make mounts read-only */
         DISSECT_IMAGE_READ_ONLY           = DISSECT_IMAGE_DEVICE_READ_ONLY |
                                             DISSECT_IMAGE_MOUNT_READ_ONLY,
+        DISSECT_IMAGE_MOUNT_IDMAPPED      = 1 << 18, /* Mount mounts with kernel 5.12-style userns ID mapping, if file sytem type doesn't support uid=/gid= */
```

Comment:
I know it's a bit of a chore, but would you mind splitting up the `nspawn: introduce --private-users-ownership=map|auto` commit and have the dissect-image (and related callers) changes as a separate, standalone commit from the nspawn changes to use it? I feel it would be clearer (and smaller), since in the future we'll use this for other features